plugins/module_utils/data_set.py

```diff
@@ -300,6 +306,85 @@ def data_set_cataloged(name):
             return True
         return False
 
+    @staticmethod
+    def get_volume_list_for_cataloged_data_set(name):
+        """Get the volume list for a cataloged dataset name.
+
+        Arguments:
+            name (str) -- The data set name to check if cataloged.
+        Returns:
+            list{str} -- A list of volumes where the dataset is cataloged.
+        """
+        name = name.upper()
+        module = AnsibleModuleHelper(argument_spec={})
+        stdin = " LISTCAT ENTRIES('{0}') ALL".format(name)
+        rc, stdout, stderr = module.run_command(
+            "mvscmdauth --pgm=idcams --sysprint=* --sysin=stdin", data=stdin
+        )
+        delimeter = 'VOLSER------------'
+        arr = stdout.split(delimeter)
+        # If a volume serial is not always of lenght 6 we could use ":x.find(' ')" here instead of that index.
+        volume_list = [x[:x.find(' ')] for x in arr[1:]]
+        return volume_list
+
+    @staticmethod
+    def delete_uncataloged_dataset(name, volumes):
+        """Delete an uncataloged dataset by specifying volumes.
+
+        Arguments:
+            name (str) -- The data set name to check if cataloged.
+            volumes (list[str]) -- The volumes the data set may reside on.
+        Returns:
+            bool -- Return code from the mvs_cmd, if 0 then it was successful.
+        """
+        # if is VSAM is_vsam(name, volumes)
+        vsam_code = 'NVR'
+        vsam_name_extension = ''
+        if DataSet.is_vsam(name, volumes):
+            vsam_code = 'VVR'
+            vsam_name_extension = '.DATA'
+        command = " DELETE {0} FILE(DD1) {1}".format(name + vsam_name_extension, vsam_code)
+        dds = dict(DD1=',vol,'.join(volumes) + ',vol')
+        rc, stdout, stderr = mvs_cmd.idcams(cmd=command, dds=dds, authorized=True)
+        if rc > 0:
+            raise DatasetDeleteError(name, rc)
+        return rc
+
+    @staticmethod
+    def attempt_to_delete_uncataloged_data_set_if_necessary(name, volumes):
+        """Attempt to delete any uncataloged dataset if exists on any volume and there is a cataloged dataset with the same name.
+
+        Arguments:
+            name (str) -- The data set name to check if cataloged.
+            volumes (list[str]) -- The volumes the data set may reside on.
+        Returns:
+            bool -- If any action was performed on the data.
+            bool -- If the dataset is still present.
+            bool -- If given the volumes list and dataset name we need to continue with deleting the dataset as usual,
+            either by cataloging it and deleting or deleting a cataloged dataset.
+        """
+        changed = False
+        present = True
+        pending_to_delete_cataloged_dataset = False
+        # Get the list of volumes that the dataset is catalogued in.
+        cataloged_volume_list = DataSet.get_volume_list_for_cataloged_data_set(name)
+        if len(cataloged_volume_list) == 0:
```

Comment:
I'd ask you to double-check if DataSet.get_volume_list_for_cataloged_data_set can not return "None".  If it can return none, you may need to check for that or len(None) would trigger an exception.  We've had so many odd issues from DataSet, I think it would be smart to double check.

Reply:
Will double check to make sure, do not remember testing that.

Reply:
Looks ok, @fernandofloresg you wrote the [function](https://github.com/ansible-collections/ibm_zos_core/blob/1a025b441bdd632d052de963d4630fc0da8bf47f/plugins/module_utils/data_set.py#L310) seems to be the behavior you expect, nice addition.